front_end/panels/ai_chat/core/AgentNodes.ts

```
@@ -421,9 +427,21 @@ export function createAgentNode(modelName: string, temperature: number): Runnabl
 }
 
 export function createToolExecutorNode(state: AgentState): Runnable<AgentState, AgentState> {
-  const tools = getAgentToolsFromState(state); // Adjusted to use getAgentToolsFromState
+  // If AgentNode has pre-selected tool names, honor that set to ensure consistency
+  const selectedNames: string[] | undefined = (state.context as any)?.selectedToolNames;
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)